lib/synced/strategies/full.rb

```diff
@@ -92,11 +92,14 @@ def remote_objects_persistor
         lambda do |remote_objects|
           additional_errors_check
           @remote_objects_ids.concat(remote_objects.map(&:id))
+          local_objects_hash = relation_scope.where(@id_key => remote_objects.map(&:id)).each_with_object({}) do |local_object, hash|
```

Comment:
Too much going on here, please assign `relation_scope.where(@id_key => remote_objects.map(&:id))` to some variable like `local_objects`